core/src/main/java/jenkins/tasks/SimpleBuildStep.java

```diff
@@ -115,6 +116,32 @@ default void perform(@NonNull Run<?, ?> run, @NonNull FilePath workspace, @NonNu
         }
     }
 
+    /**
+     * Indicates whether or not this step requires a launcher.
+     * <p>
+     * If this return {@code false}, this means that {@link #perform(Run, FilePath, EnvVars, Launcher, TaskListener)}
+     * will accept {@code null} for its {@code launcher} parameter.
+     *
+     * @return {@code true} when this step requires a launcher; {@code false} otherwise.
+     * @since TODO
+     */
+    default boolean requiresLauncher() {
+        return true;
+    }
+
+    /**
+     * Indicates whether or not this step requires a workspace.
+     * <p>
+     * If this return {@code false}, this means that {@link #perform(Run, FilePath, EnvVars, Launcher, TaskListener)}
+     * will accept {@code null} for its {@code workspace} parameter.
+     *
+     * @return {@code true} when this step requires a workspace; {@code false} otherwise.
+     * @since TODO
```

Comment:
It looks like there still 3 little open TODOs in this PR, cf. "@since TODO"

Reply:
Aside from this PR being closed, those are to be the version number of the first release that includes the API. Because this is a moving target, it is usually left to be filled in after the PR is merged.